src/sql/workbench/parts/query/common/queryResultsInput.ts

```diff
@@ -12,21 +12,48 @@ import { MessagePanelState } from 'sql/workbench/parts/query/browser/messagePane
 import { QueryPlanState } from 'sql/workbench/parts/queryPlan/electron-browser/queryPlan';
 import { ChartState } from 'sql/workbench/parts/charts/browser/chartView';
 import { TopOperationsState } from 'sql/workbench/parts/queryPlan/browser/topOperations';
+import { QueryModelViewState, QueryModelViewTab } from 'sql/workbench/parts/query/modelViewTab/queryModelViewTab';
 
 export class ResultsViewState {
 	public gridPanelState: GridPanelState = new GridPanelState();
 	public messagePanelState: MessagePanelState = new MessagePanelState();
 	public chartState: ChartState = new ChartState();
 	public queryPlanState: QueryPlanState = new QueryPlanState();
 	public topOperationsState = new TopOperationsState();
+	public dynamicModelViewTabsState: Map<string, QueryModelViewState> = new Map<string, QueryModelViewState>();
+
 	public activeTab: string;
 	public visibleTabs: Set<string> = new Set<string>();
 
+	public saveDynamicTabState(tabs: QueryModelViewTab[]): void {
```

Comment:
This relationship should be flipped, the results view state should be inserted into the tabs and the tabs should be responsible for restoreing their own state.

Reply:
@anthonydresser  could you clarify specifically what you'd like to see changed?  You think restoreDynamicTabState should be moved into the tab class?

Reply:
The QueryModelViewTab should have a state setting method that takes in the result view state object and restores it's own state.

Reply:
sure, let me take a look

Reply:
@anthonydresser I've move the state logic into QueryModelViewTab.  Please take another look.